lib/sqewer/local_connection.rb

```diff
@@ -0,0 +1,134 @@
+class Sqewer::LocalConnection < Sqewer::Connection
+  FAIL_AFTER_DELIVERIES = 10
+
+  def with_db(**k)
+    SQLite3::Database.open('sqewer-local-queue.sqlite3', **k) do |db|
+      db.busy_timeout = 5
+      return yield db
+    end
+  end
+
+  def with_readonly_db(&blk)
+    with_db(readonly: true, &blk)
+  end
+
+  def initialize(queue_url)
+    require 'sqlite3'
+    @queue_url = queue_url
+    with_db do |db|
+      db.execute("CREATE TABLE IF NOT EXISTS sqewer_messages_v1 (
+        id INTEGER PRIMARY KEY AUTOINCREMENT ,
+        queue_url VARCHAR NOT NULL,
+        receipt_handle VARCHAR NOT NULL,
+        deliver_after_epoch INTEGER,
+        times_delivered_so_far INTEGER DEFAULT 0,
+        last_delivery_at_epoch INTEGER,
+        visible BOOLEAN DEFAULT 't',
+        message_body TEXT)"
+      )
+      db.execute("CREATE INDEX IF NOT EXISTS index_sqewer_messages_v1_on_receipt_handle ON sqewer_messages_v1 (receipt_handle)")
+      db.execute("CREATE INDEX IF NOT EXISTS index_sqewer_messages_v1_on_queue_url ON sqewer_messages_v1 (queue_url)")
+    end
```

Comment:
If we decide to keep using a fixed file for the database, what happens when a previous test failed before it could delete all its items? We might want a `DELETE FROM sqewer_messages_v1` here, but it would also prevent people from preloading their db with useful test data.

Reply:
My idea for now is that you shouldn't be "snuffing around" in the dbfile itself from (say) a test suite of an application that uses Sqewer as a library, since it is an implementation detail that is internal to sqewer.